packages/grafana-data/src/utils/dataLinks.ts

```diff
@@ -88,30 +88,34 @@ function generateInternalHref<T extends DataQuery = any>(
   );
 }
 
-function interpolateObject<T extends object>(
-  object: T | undefined,
+function interpolateObject<T extends Object>(
```

Comment:
Needed to rewrite this function properly because otherwise it was hard to figure out how to properly escape quotes `"` in the variables to still produce correct JSON string.